### PR TITLE
Refactor TransactionDetails to include confirmation_time

### LINKED ISSUE #180: Wondering why we use Transaction::Confirmed and Transaction::Unconfirmed
The Rust library does not make such distinction; instead the [`TransactionDetails` struct](https://docs.rs/bdk/latest/bdk/struct.TransactionDetails.html#structfield.transaction) has a field `confirmation_time` which is either `None` if the transaction is unconfirmed or `BlockTime` if it is confirmed.

### PR DESCRIPTION
**Pull request by notmandatory**
### Description

Fixes #180 

#### All Submissions:

* [x] I've signed all my commits
* [x] I followed the [contribution guidelines](https://github.com/bitcoindevkit/bdk/blob/master/CONTRIBUTING.md)
* [x] I ran `cargo fmt` and `cargo clippy` before committing

#### New Features:

* [ ] I've added tests for the new feature
* [ ] I've added docs for the new feature
* [x] I've updated `CHANGELOG.md`

#### Bugfixes:

* [ ] This pull request breaks the existing API
* [ ] I've added tests to reproduce the issue which are now passing
* [ ] I'm linking the issue being fixed by this PR

### CODE REVIEW

**Comment by thunderbiscuit** [comment]
@notmandatory is this ready for review or do you still see it as draft? I'll take a look today if it's ready.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
   - Remove `generate_extended_key`, returned ExtendedKeyInfo [#154]
   - Remove `restore_extended_key`, returned ExtendedKeyInfo [#154]
   - Remove dictionary `ExtendedKeyInfo {mnenonic, xprv, fingerprint}` [#154]
+  - Remove interface `Transaction` [#190]
+  - Changed `Wallet` interface `list_transaction()` to return array of `TransactionDetail` [#190]
```
Comment (thunderbiscuit) [inline CHANGELOG.md:16]: Small nit: the struct is `TransactionDetails`

**Review by thunderbiscuit — APPROVED** [review]
Tested ACK. Small typo in the changelog but that's it. 🚀

**Comment by thunderbiscuit** [comment]
Rust-related question: why does confirmation_time here require to be cloned and the others don't?

```rust
impl From<&bdk::TransactionDetails> for TransactionDetails {
    fn from(x: &bdk::TransactionDetails) -> TransactionDetails {
        TransactionDetails {
            fee: x.fee,
            txid: x.txid.to_string(),
            received: x.received,
            sent: x.sent,
            confirmation_time: x.confirmation_time.clone(),
        }
    }
}
```

**Comment by notmandatory** [comment]
> Rust-related question: why does confirmation_time here require to be cloned and the others don't?

This is because the other types have the `Copy` trait, meaning they are native types that can be copied into the `TransactionDetails` automatically. But `confirmation_time` is a structure that we created so it needs to be explicitly cloned from our reference to the bdk version`bdk::TransactionDetails`.

File: CHANGELOG.md (hunk 1 above)
Comment (notmandatory) [reply CHANGELOG.md]: good catch, fixed. Also rebased.

**Pull request merged**